signingscript/src/signingscript/sign.py

```diff
@@ -137,11 +137,11 @@ def get_autograph_config(autograph_configs, cert_type, signing_formats, raise_on
 
     """
     for a in autograph_configs.get(cert_type, []):
-        if a and (set(a.formats) & set(signing_formats)):
+        if a and (set(a.formats) & set(signing_formats)) and a.key_id == keyid:
```

Comment:
I think this breaks e.g. xpi dep signing, where the keyid is in the autograph config but not in the request.

Might need something like `(keyid is None or a.key_id == keyid)` if we don't want to reshuffle the config too much?

Reply:
Good catch! Though why wouldn't it be in the request? Just because it works?

I wonder if it would be worth forcing requests to be explicit about keyids in the future. Definitely not a problem to solve in this PR though.

Reply:
Historically, we've had "default" keyids for each format IIRC. I think we may have only support 1 per format for awhile. I'm totally onboard with being explicit here in the future. (Aki might have some historical perspective to add here as well.)

Reply:
Taking a step back, trying to sum up how this works, as far as I can tell...

The tasks can specify two things:
- scopes (typically dep/nightly/release)
- a set of "formats" for each file to sign
Those two things then map (via config) to an autograph url, credentials, format and optional key id.  The format can optionally include a key id that currently overrides the one set in the config.

The key id is sent to autograph, and it lets a single username (hawk_id?) use different keys, so AIUI that's what it's for in the xpi case.  By default if we don't set the key id, the username is the key id (I think?).

Here if I understand correctly we currently have a single key id for the ev user, so we don't need to pass the key id to autograph (although we probably can if we want to).  One reason to use a keyid in signingscript is to let us reuse `sign_authenticode_file` while using a different cert and CA than it currently does.  That looks like it requires setting the key id in the task payload though, so that's maybe not great when we need to rotate keys.  I'll ponder some more and think of possible alternatives...

Reply:
Wouldn't we just rotate the key in-place in `signingscript` and leave the keyid the same?

Instead of using an `ev` keyid, should I instead create a new signing format (that still uses `sign_authenticode_file`), but then have some hardcoded logic that says:
```
if signing_format == "x":
    use this key
else:
    use that key
```
?

I thought the keyid approach was cleaner, but maybe it's using it for something that wasn't intended?

Reply:
> Instead of using an `ev` keyid, should I instead create a new signing format (that still uses `sign_authenticode_file`), but then have some hardcoded logic that says:
> 
> ```
> if signing_format == "x":
>     use this key
> else:
>     use that key
> ```
> 
> ?

That seems like the easiest way forward at this point.

> I thought the keyid approach was cleaner, but maybe it's using it for something that wasn't intended?

Agreed, it seemed promising but that doesn't seem to quite do what we want here :/